### PR TITLE
Strengthen `connectors/storage` tests and fix parsing edge cases

### PR DESCRIPTION
**Pull request by kynzie**
This PR strengthens the test suite for `connectors/storage` with more focused black-box coverage.

It also fixes a few edge cases uncovered while expanding the tests.

## Changes

- add focused tests for:
  - `Serialize`
  - `Deserialize`
  - `NewConfiguration`
  - `(*Configuration).ToBytes`
  - `NewConfigurationFromBytes`
  - `NewConfigurationFromWrappedBytes`
  - `Register`
  - `Unregister`
  - `Backends`
  - `New`
  - `Open`
  - `Create`
 
## Bug fixes

This PR also fixes a few issues uncovered by the new tests:

- handle truncated footers correctly in `deserializeReader.Read()`
- return the correct byte count on HMAC mismatch in `deserializeReader.Read()`
- make `NewConfigurationFromWrappedBytes()` parse wrapped data safely instead of assuming the input is long enough and well-formed

### Before 

```
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:37:	newDeserializeReader				0.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:65:	Read								0.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:123:	Close								0.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:127:	Deserialize							0.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:144:	newSerializeReader					0.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:162:	Read								0.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:225:	Serialize							0.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:46:			init								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:68:			NewConfiguration					100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:83:			NewConfigurationFromBytes			0.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:93:			NewConfigurationFromWrappedBytes	0.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:109:			ToBytes								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:131:			String								0.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:179:			Register							66.7%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:186:			Unregister							0.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:193:			Backends							100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:197:			New									57.1%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:223:			Open								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:238:			Create								100.0%
total:									(statements)														19.2%
```

### After

```
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:37:	newDeserializeReader				100.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:65:	Read								100.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:127:	Close								100.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:131:	Deserialize							100.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:148:	newSerializeReader					100.0%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:166:	Read								96.8%
github.com/PlakarKorp/kloset/connectors/storage/serialization.go:229:	Serialize							100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:46:			init								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:68:			NewConfiguration					100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:83:			NewConfigurationFromBytes			100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:93:			NewConfigurationFromWrappedBytes	90.9%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:117:			ToBytes								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:139:			String								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:187:			Register							100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:194:			Unregister							100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:201:			Backends							100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:205:			New									100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:231:			Open								100.0%
github.com/PlakarKorp/kloset/connectors/storage/storage.go:246:			Create								100.0%
total:									(statements)														98.7%
```

### CODE REVIEW

File: connectors/storage/storage.go [hunk 1]
```diff
@@ -91,19 +91,27 @@ func NewConfigurationFromBytes(version versioning.Version, data []byte) (*Config
...
-	err := msgpack.Unmarshal(data, &configuration)
+	payload, err := io.ReadAll(reader)
 	if err != nil {
 		return nil, err
 	}
-	configuration.Version = version
-	return &configuration, nil
+
+	return NewConfigurationFromBytes(version, payload)
 }
```
Comment (omar-polo) [inline connectors/storage/storage.go:115]: need to think a bit more about this change, but the rest is fine by me.

**Review by omar-polo — APPROVED** [review]

**Review by poolpOrg — APPROVED** [review]
okie

**Review by mathieu-plak — APPROVED** [review]
